experiment/generate_tests.py

```diff
@@ -0,0 +1,233 @@
+#!/usr/bin/env python
+
+# Copyright 2017 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Create e2e test definitions.
+
+Usage example:
+
+  In $GOPATH/src/k8s.io/test-infra,
+
+  $ ./experiment/generate_tests.py \
+      --yaml-config-path=experiment/test_config.yaml \
+      --json-config-path=jobs/config.json
+"""
+
+import argparse
+import hashlib
+import json
+import os
+import yaml
+
+# pylint: disable=too-many-branches,too-many-statements,too-many-locals
+
+
+# TODO(yguo0905): Generate Prow and testgrid configurations.
+
+def get_sha1_hash(data):
+    sha1_hash = hashlib.sha1()
+    sha1_hash.update(data)
+    return sha1_hash.hexdigest()
+
+
+def substitute(job_name, lines):
+    return [line.replace('${job_name_hash}', get_sha1_hash(job_name)[:10]) \
+            for line in lines]
+
+
+def get_envs(desc, field):
+    """Returns a list of envs from the given field, or an empty list if the
+    field is None or the field does not contain the key "envs"."""
+
+    result = ['', '# The %s configurations.' % desc]
+    if field is None:
```

Comment:
Please use an implicit false evaluation (`if not field`) or else specify why you need to distinguish `None` from `''`

https://google.github.io/styleguide/pyguide.html?showone=True/False_evaluations#True/False_evaluations

Please scan this whole file for `None` and replace them